docs/DafnyRef/UserGuide.md

```diff
@@ -315,7 +315,260 @@ This list is not exhaustive but can definitely be useful to provide the next ste
   <br><br>`assert forall i | i == m :: P(m);` |  `assert P(m);`<br>`assert forall i | i == m :: P(i);`
   `method m(i) returns (j: T)`<br>&nbsp;&nbsp;`  requires A(i)`<br>&nbsp;&nbsp;`  ensures B(i, j)`<br>`{`<br>&nbsp;&nbsp;`  ...`<br>`}`<br><br>`method n() {`<br>&nbsp;&nbsp;`  ...`<br><br><br>&nbsp;&nbsp;`  var x := m(a);`<br>&nbsp;&nbsp;`  assert P(x);` | `method m(i) returns (j: T)`<br>&nbsp;&nbsp;`  requires A(i)`<br>&nbsp;&nbsp;`  ensures B(i, j)`<br>`{`<br>&nbsp;&nbsp;`  ...`<br>`}`<br><br>`method n() {`<br>&nbsp;&nbsp;`  ...`<br>&nbsp;&nbsp;`  assert A(k);`<br>&nbsp;&nbsp;`  assert forall x :: B(k, x) ==> P(x);`<br>&nbsp;&nbsp;`  var x := m(k);`<br>&nbsp;&nbsp;`  assert P(x);`
 
-### 24.8.2. Assertion batches {#sec-assertion-batches}
+### 24.8.2. Verification debugging when verification is slow {#sec-verification-debugging-slow}
+
+In this section, we describe methodologies to apply in the case when verification is slower than expected, does not terminate, or timeouts.
```

Comment:
How about a first subsection that suggests verifying with more, smaller assertion batches. Personally I would suggest first trying one batch per assertion before moving on to the techniques below. You can link the reader to https://dafny-lang.github.io/dafny/DafnyRef/DafnyRef#sec-assertion-batches for more details.